OIPA/iati/parser/tests/test_2_03_parser.py

```diff
@@ -2926,6 +2926,138 @@ def test_activity_result_indicator_baseline(self):
+
+        dummy_source = synchroniser_factory.DatasetFactory.create()
+
+        self.parser_203 = ParseManager(
+            dataset=dummy_source,
+            root=self.iati_203_XML_file,
+        ).get_parser()
+
+        self.result_indicator = iati_factory.ResultIndicatorFactory()
+
+        self.result_indicator_baseline = iati_factory.\
+            ResultIndicatorBaselineFactory(
+                result_indicator=self.result_indicator,
+            )
+
+        self.activity = self.result_indicator_baseline.result_indicator\
+            .result.activity
+
+        self.location = iati_factory.LocationFactory(ref='KH-PNH')
+
+        self.parser_203.register_model('Activity', self.activity)
+
+        self.parser_203.register_model(
+            'ResultIndicatorBaseline',
+            self.result_indicator_baseline
+        )
+        self.parser_203.register_model('Location', self.location)
+
+    def test_activity_result_indicator_baseline_location(self):
+
+        """
+        Test if <location> element in context of a baseline element (as part
+        of a parent result/indicator element) is parsed and saved correctly
+        """
+
+        # 1. 'ref' attribute is not provided:
+        result_indicator_baseline_location_attrs = {
+            # 'ref': 'AF-KAN',
+        }
+
+        result_indicator_baseline_location_XML_element = E(
+            'location',
+            **result_indicator_baseline_location_attrs
+        )
+
+        try:
+            self.parser_203\
+                .iati_activities__iati_activity__result__indicator__baseline__location(  # NOQA: E501
+                    result_indicator_baseline_location_XML_element
+                )
+        except RequiredFieldError as inst:
+            self.assertEqual(inst.field, 'ref')
+            self.assertEqual(inst.message,
+                             "This attribute has to be a cross-reference to "
+                             "the internal reference assigned to a defined "
+                             "location: iati-activity/location/@ref, so "
+                             "leaving it blank makes no sense")
+
+        # 2. Referenced location doesn't exist:
+        result_indicator_baseline_location_attrs = {
+            'ref': 'AF-KAN',  # it's not 'KH-PNH'
+        }
+
+        result_indicator_baseline_location_XML_element = E(
+            'location',
+            **result_indicator_baseline_location_attrs
+        )
+
+        try:
+            self.parser_203\
+                .iati_activities__iati_activity__result__indicator__baseline__location(  # NOQA: E501
+                    result_indicator_baseline_location_XML_element
+                )
+        except FieldValidationError as inst:
+            self.assertEqual(inst.field, 'ref')
+
+            self.assertEqual(
+                inst.message,
+                "Referenced location doesn't exist"
+            )
+
+        # 3. Referrenced location isn't refferenced on an activity level:
```

Comment:
Just trivial spelling error. It should be 'Referenced'

Reply:
Fixed with 73b221bb9c2b9714c96f8cbb926a37ffee7d6f9d